machine/v1alpha1/provider_types.go

```diff
@@ -0,0 +1,207 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrlconfigv1 "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	ProviderFinalizer         = "provider.cluster.x-k8s.io"
+	ConfigMapVersionLabelName = "provider.cluster.x-k8s.io/version"
+)
+
+// ProviderSpec is the desired state of the Provider.
+type ProviderSpec struct {
+	// Version indicates the provider version.
+	Version string `json:"version,omitempty"`
+
+	// Manager defines the properties that can be enabled on the controller manager for the provider.
+	// +optional
+	Manager *ManagerSpec `json:"manager,omitempty"`
+
+	// Deployment defines the properties that can be enabled on the deployment for the provider.
+	// +optional
+	Deployment *DeploymentSpec `json:"deployment,omitempty"`
+
+	// SecretName is the name of the Secret providing the configuration
+	// variables for the current provider instance, like e.g. credentials.
+	// Such configurations will be used when creating or upgrading provider components.
+	// The contents of the secret will be treated as immutable. If changes need
+	// to be made, a new object can be created and the name should be updated.
+	// The contents should be in the form of key:value. This secret must be in
+	// the same namespace as the provider.
+	// +optional
+	SecretName *string `json:"secretName,omitempty"`
+
+	// FetchConfig determines how the operator will fetch the components and metadata for the provider.
+	// If nil, the operator will try to fetch components according to default
+	// embedded fetch configuration for the given kind and `ObjectMeta.Name`.
+	// For example, the infrastructure name `aws` will fetch artifacts from
+	// https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases.
+	// +optional
+	FetchConfig *FetchConfiguration `json:"fetchConfig,omitempty"`
+}
+
+// ManagerSpec defines the properties that can be enabled on the controller manager for the provider.
+type ManagerSpec struct {
+	// ControllerManagerConfigurationSpec defines the desired state of GenericControllerManagerConfiguration.
+	ctrlconfigv1.ControllerManagerConfigurationSpec `json:",inline"`
+
+	// ProfilerAddress defines the bind address to expose the pprof profiler (e.g. localhost:6060).
+	// Default empty, meaning the profiler is disabled.
+	// Controller Manager flag is --profiler-address.
+	// +optional
+	ProfilerAddress *string `json:"profilerAddress,omitempty"`
+
+	// MaxConcurrentReconciles is the maximum number of concurrent Reconciles
+	// which can be run. Defaults to 10.
+	// +optional
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=1
+	MaxConcurrentReconciles *int `json:"maxConcurrentReconciles,omitempty"`
+
+	// Verbosity set the logs verbosity. Defaults to 1.
+	// Controller Manager flag is --verbosity.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	Verbosity int `json:"verbosity,omitempty"`
```

Comment:
How does a default differentiate between an explicit zero and not setting the field? Have you tested this at all?

Reply:
https://github.com/kubernetes-sigs/cluster-api-operator/blob/main/controllers/component_customizer.go#L186

Reply:
I don't think this answered my question, if 0 is a valid value, then a user which sets `Verbosity: 0` will have the value omitted (because this is optional), so then it will be defaulted to `1` again. If 0 is a valid value, this needs to be a pointer so there is a differentiation between a zero value and a `0` value